mozilla/mozillafestival.org
# Submit proposal to Google Spreadsheet.

Related to #569 - proposal page.
Fixes #546 - nav bar and footer changes.

Some copy and links are missing - I will update them later when them are finalized!

---

Hey @Pomax we are under tight timeline, please help me review and spot bugs that will prevent a successful submission flow. For UI/UX improvement let's file follow-up tickets!

### Expected flow

This is the first part of the proposal flow: **take proposal from the MozFest to a Google Spreadsheet**.
After this is merged I'll work on 
1. getting GitHub tickets filed upon form submission (example "proposal" tickets see https://github.com/MozillaFoundation/mozfest-program-2016/issues). This work will be implemented in another PR.
2. sending out transaction email to the submitter to tell him/her to check out the GitHub (proposal) repo

**Take proposal from the MozFest to a Google Spreadsheet**
1. user to go `/proposals` page
2. fill out all required form and submit it (form requirements are based off of this [Google Form](https://docs.google.com/a/mozillafoundation.org/forms/d/10Pwt_tVkqvxqnz6KTV3Wf2HDz5WQCq78toQKB5E8pXM/edit?ts=59258a06)) (used as reference)
3. proposal gets submitted to a Google Spreadsheet
4. page should update upon form submission (success or fail)

### To test

Three new required env vars:

(instructions on how to get a Google Service Account: https://www.npmjs.com/package/google-spreadsheet#service-account-recommended-method)

- `PROPOSAL_SPREADSHEET_ID_2017`
  - you can probably use the spreadsheet I created for testing: `16QMKfetQbW7N0a21URIYoV16ZtFPj8bqOCujVUV-oIc`
- `GOOGLE_API_CLIENT_EMAIL_2017`: your Google Service Account client email
- `GOOGLE_API_PRIVATE_KEY_2017`: your Google Service Account private key

Also make sure you give your Google Service account **write access** to this testing spreadsheet: https://docs.google.com/spreadsheets/d/16QMKfetQbW7N0a21URIYoV16ZtFPj8bqOCujVUV-oIc/edit#gid=0
